### PR TITLE
Add the Transformer module and its corresponding unittest

### PR DESCRIPTION
**Pull request by eleven-yun**

### CODE REVIEW

File: nn_modules/transformer/top_former.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import os
+import sys
+import torch.nn as nn
+
+cwd = os.getcwd()  # current working directory
```
Comment (LingerDavid) [inline nn_modules/transformer/top_former.py:5]: Other solution to replace?

File: nn_modules/transformer/top_former.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            The dimension of input/output feature representation.
+        nhead : int
+            The number of attention heads.
+        num_classes : int
+            The number of classes.
+
+        """
+        super(TopFormer, self).__init__()
+        self.encoder = TopEncoder(num_layers, d_model, nhead)
+        self.decoder = TopDecoder(num_layers, d_model, nhead)
```
Comment (LingerDavid) [inline nn_modules/transformer/top_former.py:32]: How to make use of the feature from the decoder?

File: nn_modules/transformer/top_former.py (hunk 1 above)
Comment (LingerDavid) [reply nn_modules/transformer/top_former.py:5]: Try to add `__init__.py` to solve.

Comment (eleven-yun) [reply nn_modules/transformer/top_former.py:5]: To be addressed by a follow-up patch.

**Review by LingerDavid — APPROVED** [review]

**Pull request merged**